ocaml/lambda/lambda.ml

```diff
@@ -217,6 +217,7 @@ type primitive =
   | Plsrbint of boxed_integer * alloc_mode
   | Pasrbint of boxed_integer * alloc_mode
   | Pbintcomp of boxed_integer * integer_comparison
+  | Punboxed_int_comp of boxed_integer * integer_comparison
```

Comment:
It's a bit surprising that a constructor with "unboxed" in the name would carry a `boxed_integer` payload. Is this correct? If so, do you think it's worth commenting this?

FWIW, I'd be in favor of introducing a new type `unboxed_integer` that has the same constructors as what's currently in `boxed_integer`, if that's easy. It could even be a private alias.

Reply:
I think that already exists in the unboxed arrays PR actually.

Reply:
I thought I'd added such a type though, maybe not...

Reply:
actually, you did (see https://github.com/ocaml-flambda/flambda-backend/pull/2185/files#diff-4e8efb33bb39fe1c7f2c33b3747200c58d90d8c8ba21956fb7ecc52403b9b2d8R329 ), but as an alias of `boxed_integer` which is a bit weird the first time one reads the type definition.

Reply:
For compatibility with the other PR, I adopted the same approach here:

- defined `unboxed_integer` as an alias to `boxed_integer`
- and changed `Punboxed_int_comp` to take `unboxed_integer`

Reply:
OK. I think it would be a bit better to mint another type with the same constructors, but I'm happy to leave this as-is until after the other PR is merged as well. (This admits that we may never fix this, but I don't think the state is too bad.)